yaml2sbml/YamlModel.py

```diff
@@ -0,0 +1,589 @@
+import yaml
+import os.path
+from typing import Union
+
+from .yaml2sbml import _parse_yaml_dict
+from .yaml2PEtab import _yaml2petab
+from .yaml_validation import _validate_yaml_from_dict
+
+
+class YamlModel:
+    """
+    Functionality to set up, edit, load and write yaml models.
+    """
+
+    def __init__(self):
```

Comment:
why not initializing with all the dictionary keys (observables, variables, time, etc.) ?

Reply:
If I would do this, I would often have empty keys (e.g. `self._yaml_model['assignments']=[]`). If I then write the model to yaml, I would get empty blocks, like below. When you do not even have these keys, you can avoid that. :)
```yaml
assignments:

observables: 

```

Reply:
I can add a check for the `model.delete_[...]` functions, to test if I should delete the key afterwards :)

Reply:
aaah! that makes sense, but yeah, as you are pointing out i think, we can just check and delete empty keys when writing the model to yaml/sbml/petab.
the main reason why it makes sense in my head to initialize all the other keys is to have a more clear picture of what the model object looks like, which attributes does it actually have. thoughts?

(just to be clear, at the end of the day i'm not going to block wtv you think makes more sense :) )

Reply:
That makes sense and is a cleaner solution :) I will do it that way :)
(and no worries ;) quite the opposite: Thank you for the feedback :) )